### PR TITLE
[APPSEC-793] Specify misc top 50 secrets

### PR DESCRIPTION
**Pull request by gaetan-ferry-sonarsource**

### CODE REVIEW

**Review by jamie-anderson-sonarsource — CHANGES_REQUESTED** [review]

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/django.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          match: django-insecure-r&lvybzry1*k+qq)=x-!=0yd5l5#1gxzk!82@ru25*ntos3_9^
+
+    - id: settings.py-secret-key
+      metadata:
+        name: Django secret keys should not be disclosed
+      detection:
+        pre:
+          include:
+            paths:
+              - "*/settings.py"
```
[the remark this answers is not in the document]
Comment (gaetan-ferry-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/django.yaml:31]: Agreed. Considering the unspecific type of secret here, I'd rather not try to cover a lot of corner cases.

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/spotify.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    message: Make sure this Spotify key gets revoked, changed, and removed from the code.
+  detection:
+    pre:
+      include:
+        content:
+          - "spotify"
+  rules:
+    - id: spotify-api-secret
+      metadata:
+        name: OpenWeather API keys should not be disclosed
```
[the remark this answers is not in the document]
Comment (gaetan-ferry-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/spotify.yaml:14]: oopsy

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/django.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          pattern: "(?m)\\bSECRET_KEY(?:_FALLBACKS)?\\s*=\\s*(.*)$"
+          context:
+            matchEither:
+              - patternAround: "(?i)\\bdjango\\b"
+              - patternAround: "ALLOWED_HOSTS\\s*=" # Chosen because has the most chances of being configured per env
+        post:
+          heuristicFilter:
+            heuristics:
+              - "path"
+          patternNot: "(?i)({{.*?}}|sample|example|x{4,}|0{4,})"
```
Comment (jamie-anderson-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/django.yaml:66]: The opening curly braces near the start of the pattern need to be escaped. It appears that the closing braces are ok, at least according to Regex101.

```suggestion
          patternNot: "(?i)(\\{\\{.*?}}|sample|example|x{4,}|0{4,})"
```

**Review by jamie-anderson-sonarsource — CHANGES_REQUESTED** [review]

**Review by jamie-anderson-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**